presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkModule.java

```diff
@@ -478,6 +485,8 @@ protected void setup(Binder binder)
 
         binder.bind(RemoteTaskStats.class).in(Scopes.SINGLETON);
         newExporter(binder).export(RemoteTaskStats.class).withGeneratedName();
+        binder.bind((new TypeLiteral<PrestoSparkShuffleInfoFactory<?, ?>>() {})).to(PrestoSparkLocalShuffleInfoFactory.class).in(Scopes.SINGLETON);
```

Comment:
So here we use the binder to flexibly bind oss and non-oss shuffle info factory?

Reply:
You're right, we'll bind a Cosco shuffleInfo factory for Cosco shuffle.